src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationExtensions.cs

```diff
@@ -42,6 +41,8 @@ internal static PEMethodSymbol GetSourceMethod(this CSharpCompilation compilatio
                         var module = metadataDecoder.Module;
                         methodHandle = candidateMethod.Handle;
                         string stateMachineTypeName;
+
+                        if (GeneratedNames.GetKind(containingType.Name) == GeneratedNameKind.StateMachineType)
```

Comment:
Is this check needed? If so, please move before the `foreach`.

Reply:
No, it is not needed. Sorry! Thanks for the catch!

---
In reply to: [135918337](https://github.com/dotnet/roslyn/pull/21707#discussion_r135918337) [](ancestors = 135918337)